lib/pbench/server/auth/auth.py

```diff
@@ -1,179 +1,183 @@
-                    e,
-                )
-        except jwt.InvalidTokenError:
-            pass  # Ignore this silently; client is unauthenticated
-        except Exception as e:
-            Auth.logger.exception(
-                "Unexpected exception occurred while verifying the auth token {!r}: {}",
-                auth_token,
-                e,
+                message="Malformed Authorization header, request needs bearer token: Bearer <session_token>",
             )
-        return None
-
-    @staticmethod
-    def verify_third_party_token(auth_token: str, oidc_client: OpenIDClient) -> bool:
-        """
-        Verify a token provided to the Pbench server which was obtained from a
-        third party identity provider.
-        Args:
-            auth_token: Token to authenticate
-            oidc_client: OIDC client to call to authenticate the token
-        Returns:
-            True if the verification succeeds else False
-        """
-        try:
-            oidc_client.token_introspect_offline(
-                token=auth_token,
-                key=oidc_client.get_oidc_public_key(),
-                audience=oidc_client.client_id,
-                options={
-                    "verify_signature": True,
-                    "verify_aud": True,
-                    "verify_exp": True,
-                },
-            )
-            return True
-        except (
-            jwt.ExpiredSignatureError,
-            jwt.InvalidTokenError,
-            jwt.InvalidAudienceError,
-        ):
-            Auth.logger.error("OIDC token verification failed")
-            return False
-        except Exception:
-            Auth.logger.exception(
-                "Unexpected exception occurred while verifying the auth token {}",
-                auth_token,
-            )
-
-        if not oidc_client.TOKENINFO_ENDPOINT:
-            Auth.logger.warning("Can not perform OIDC online token verification")
-            return False
-
-        try:
-            token_payload = oidc_client.token_introspect_online(
-                token=auth_token, token_info_uri=oidc_client.TOKENINFO_ENDPOINT
-            )
-        except OpenIDClientError:
-            return False
-
-        return "aud" in token_payload and oidc_client.client_id in token_payload["aud"]
+        return auth_token
+
+
+def verify_token_only(auth_token: str) -> str:
+    """Returns "verified", "expired", or "invalid" depending on the state of the
+    given token.
+    """
+    try:
+        jwt.decode(
+            auth_token,
+            current_app.secret_key,
+            algorithms=_TOKEN_ALG,
+            options={
+                "verify_signature": True,
+                "verify_aud": False,
+                "verify_exp": True,
+            },
+        )
+    except jwt.InvalidTokenError:
+        state = "invalid"
+    except jwt.ExpiredSignatureError:
+        state = "expired"
+    else:
+        state = "verified"
+    return state
```

Comment:
I know your usage is limited, but I'd still feel happier if these were Enums rather than just strings.